src/Sulu/Bundle/HttpCacheBundle/CacheLifetime/CacheLifetimeEnhancer.php

```diff
@@ -33,14 +33,21 @@ class CacheLifetimeEnhancer implements CacheLifetimeEnhancerInterface
      */
     private $sharedMaxAge;
 
+    /**
+     * @var CacheLifetimeRequestEnhancer
```

Comment:
This sounds really similar to the `CacheLifetimeEnhancer`. I think using something like `RequestCacheLifetimeStore` would be easier to distinguish

Reply:
or maybe add a `addMaxCacheLifetime` method to this class?

Reply:
Still don't think that it is a `Store` though... But happy to change it, if you think it makes more sense.

Regarding the method name, I don't think that `addMaxCacheLifetime` is a good one... `add` seems like a non-destructive operation, so nothing that has already been there is lost, but this is not the case... `set` might not be perfect either, but IMO it's closer. Something like `setCacheLifetimeIfShorter` would be the clearest, but it does not feel very nice to me, and it couples the name of the function to the implementation.

Reply:
No really happy with `store` too. But I dont have a better idea and i think it makes things easier to understand. So i would change it 🙂

Reply:
I think `Enhancer` was easier to understand, because it is also kind of related to the `RouteEnhancer`s from the `CmfRoutingBundle`. But if we rename it, I would go with `CacheLifetimeRequestStore`. When the words `Request` and `Store` are closer together I would think it is more likely to assume that the information is stored on the request.

Reply:
> I think Enhancer was easier to understand, because it is also kind of related to the RouteEnhancers from the CmfRoutingBundle

If there wasnt already a `CacheLifetimeEnhancer`, `CacheLifetimeRequestEnhancer` would be good for me too. But we already have a `CacheLifetimeEnhancer` that does something different, therefore I would prefer`CacheLifetimeRequestStore` 🙂  Yes, `CacheLifetimeRequestStore::setCacheLifetime` is okay for me